demo/src/main/java/net/minestom/demo/PlayerInit.java

```diff
@@ -187,24 +71,145 @@ public class PlayerInit {
         //     System.out.println("load end");
         // });
 
-        inventory = new Inventory(InventoryType.CHEST_1_ROW, Component.text("Test inventory"));
+        inventory = new Inventory(serverFacade.getGlobalEventHandler(), serverFacade.getServerSettings(), InventoryType.CHEST_1_ROW, Component.text("Test inventory"));
         inventory.setItemStack(3, ItemStack.of(Material.DIAMOND, 34));
+
+        DEMO_NODE = EventNode.all(serverFacade, "demo")
+                .addListener(EntityAttackEvent.class, event -> {
+                    final Entity source = event.getEntity();
+                    final Entity entity = event.getTarget();
+
+                    entity.takeKnockback(0.4f, Math.sin(source.getPosition().yaw() * 0.017453292), -Math.cos(source.getPosition().yaw() * 0.017453292));
+
+                    if (entity instanceof Player) {
+                        Player target = (Player) entity;
+                        target.damage(Damage.fromEntity(source, 5));
+                    }
+
+                    if (source instanceof Player) {
+                        ((Player) source).sendMessage("You attacked something!");
+                    }
+                })
+                .addListener(PlayerDeathEvent.class, event -> event.setChatMessage(Component.text("custom death message")))
+                .addListener(PickupItemEvent.class, event -> {
+                    final Entity entity = event.getLivingEntity();
+                    if (entity instanceof Player) {
+                        // Cancel event if player does not have enough inventory space
+                        final ItemStack itemStack = event.getItemEntity().getItemStack();
+                        event.setCancelled(!((Player) entity).getInventory().addItemStack(itemStack));
+                    }
+                })
+                .addListener(ItemDropEvent.class, event -> {
+                    final Player player = event.getPlayer();
+                    ItemStack droppedItem = event.getItemStack();
+
+                    Pos playerPos = player.getPosition();
+                    ItemEntity itemEntity = new ItemEntity(serverFacade, droppedItem);
```

Comment:
There are a lot of odd cases created now because you could try to spawn an entity in a level of server A while passing server B into the constructor. These probably need to be handled somewhere and I'm not sure they are (im still on like file 2 of review)

Reply:
In theory, two different servers can have common components, for example a common ExceptionHandler